source/PlayerInfo.cpp

```diff
@@ -1398,16 +1401,21 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 
 // Find out how attractive the player's fleet is to pirates. Aside from a
 // heavy freighter, no single ship should attract extra pirate attention.
-pair<double, double> PlayerInfo::RaidFleetFactors() const
+pair<double, double> PlayerInfo::RaidFleetFactors(const System *system) const
 {
 	double attraction = 0.;
 	double deterrence = 0.;
+	bool defaultAttraction = !system || !system->GetRaiders() || !system->GetRaiders()->ScoutsCargo();
+	double emptyCargoAttraction = defaultAttraction ? 1. : system->GetRaiders()->EmptyCargoAttraction();
 	for(const shared_ptr<Ship> &ship : Ships())
 	{
 		if(ship->IsParked() || ship->IsDestroyed())
 			continue;
 
-		attraction += ship->Attraction();
+		if(defaultAttraction)
+			attraction += ship->Attraction();
+		else
+			attraction += ship->Attraction() * emptyCargoAttraction * ship->Cargo().Free() / ship->Cargo().Size();
```

Comment:
Division by zero is undefined behaviour; you need to handle the case where the cargo hold has zero capacity.
Also, this looks like, if the cargo hold is full, it'll add zero to the total attraction?
You're increasing total attraction by: `x * y * (unusedCargoSpace / totalCargoSpace)`.
If all the cargo space is used, `unusedCargoSpace` is 0, so you have `x * y * 0 / totalCargoSpace` which is just zero.
That doesn't feel like what we want to be doing to me.

Reply:
thx, logic should be good now